### PR TITLE
docs(mesh): update version support table with kong mesh 2.1.x

### PR DESCRIPTION
**Pull request by slonka**
### Description

What did you change and why?

We released Kuma 2.1.x on 1st of Feb so we need to update the support table. You can check the date here https://kuma.io/blog/2023/kuma-2-1-0/ .
 
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.

### Testing instructions

Netlify link: https://deploy-preview-5220--kongdocs.netlify.app/mesh/latest/support-policy/#version-support-for-kong-mesh

### Checklist 

- [X] Review label added 
- [X] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Review by bartsmykla — APPROVED** [review]

File: app/_src/mesh/support-policy.md
Comment (slonka) [reply app/_src/mesh/support-policy.md:15]: Sure

Comment (slonka) [reply app/_src/mesh/support-policy.md:15]: Is 2.0.x correct? It has 2yrs of support and previous releases had 1 year. Or is 2.0.x special and 2.1.x should have 1 year?

Comment (slonka) [reply app/_src/mesh/support-policy.md:15]: @lahabana ☝️ any input on this?

**Review by cloudjumpercat — APPROVED** [review]
LGTM! Thanks for adding these! I'll hold off on merging until @lahabana approves the PR.

File: app/_src/mesh/support-policy.md
Comment (lahabana) [reply app/_src/mesh/support-policy.md:15]: You are right it should be 2023 it's only 1 year (same for 2.1)

**Review by lahabana — APPROVED** [review]
But we should fix support for 2.0 and 2.1 as they are only supported 1 year

File: app/_src/mesh/support-policy.md
Comment (cloudjumpercat) [inline app/_src/mesh/support-policy.md:15]: @slonka Could you update 2.0 and 2.1 so the support is for only one year?

File: app/_src/mesh/support-policy.md
Comment (cloudjumpercat) [inline app/_src/mesh/support-policy.md:14]:
```suggestion
|  2.1.x   |  2023-02-01   |     2024-02-01      |
```

File: app/_src/mesh/support-policy.md
Comment (slonka) [reply app/_src/mesh/support-policy.md:15]: yeah, sorry for the late reply but I was on duty and had a ton of stuff to work on, thanks! 👍

**Pull request merged**
